hydra-node/src/Hydra/Node.hs

```diff
@@ -106,27 +106,46 @@ runHydraNode tracer node@HydraNode{eq, env = Environment{party}} = do
   forever $ do
     e <- nextEvent eq
     traceWith tracer $ ProcessingEvent party e
-    processNextEvent node e >>= \case
+    atomically (processNextEvent node e) >>= \case
       -- TODO(SN): Handling of 'Left' is untested, i.e. the fact that it only
       -- does trace and not throw!
       Left err -> traceWith tracer (ErrorHandlingEvent party e err)
       Right effs -> forM_ effs (processEffect node tracer) >> traceWith tracer (ProcessedEvent party e)
 
--- | Monadic interface around 'Hydra.Logic.update'.
-processNextEvent ::
+-- | "Prepare" a node by feeding it a list of 'Event's.
+-- The events are interpreted but their effects, if any, are discarded hence the
+-- only side-effect of this function is to change the 'HeadState'. A
+-- 'LogicError' exception is thrown if events generate such errors, which leaves
+-- the 'HeadState' in its initial state as the 'STM' transaction is rolled back.
+--
+-- As this function actually runs in the 'STM' monad, it does not change the
+-- state of the head until called with 'atomically'.
+feedEvents ::
   ( Tx tx
   , MonadSTM m
+  , MonadThrow (STM m)
   ) =>
   HydraNode tx m ->
+  [Event tx] ->
+  STM m (HydraNode tx m)
+feedEvents node events = do
+  (errs, _) <- partitionEithers <$> traverse (processNextEvent node) events
+  if null errs
+    then pure node
+    else throwSTM $ MultipleErrors errs
```

Comment:
The idea behind `MultipleErrors` is to be able to inspect the errors after the fact. Right now, we do nothing with those errors, they are just traced and we keep going. So it's possible to have a Node producing a trace with several events leading to `LogicError` being returned, but when reimporting this trace we might not want to consider it valid. I think it would be better if `MultipleErrors` returned also the "final" state for further inspection.
I understand this is all not very convincing...